src/main/java/qna/domain/Answer.java

```diff
@@ -48,12 +49,16 @@ public Answer setDeleted(boolean deleted) {
         return this;
     }
 
-    public boolean isDeleted() {
-        return deleted;
+    public DeleteHistory deleteAnswerSoftly(User questionUser) throws CannotDeleteException {
+        if (writer.isOtherUser(questionUser)) {
+            throw new CannotDeleteException("다른 사람이 쓴 답변이 있어 삭제할 수 없습니다.");
```

Comment:
다음처럼 쓰는게 메시지에 적절할 것 같은데, 어떻게 생각하시나요?
```suggestion
            throw new CannotDeleteException("답변자가 아닌 사용자는 답변을 삭제할 수 없습니다.");
```
예외 메시지에는 [이펙티브 자바 Item 75. 예외의 상세 메시지에 실패 관련 정보를 담아라](https://songiam.tistory.com/81) 를 적용해봐도 좋겠네요.

Reply:
제안해주신 메세지가 더욱 실패 원인에 대해 명확히 전달할 수 있을 것 같습니다!
피드백 주셔서 감사합니다 😃